File: lib/od/identity/od_type.py
from enum import Enum


class ODType(Enum):
    """ODType enum class holds the names of all supported ODs"""
    GENERIC = "Generic",
    GO_INDEX = "GoIndex",
    FODI = "FODI"
    ZFILE = "ZFile"
    GD_INDEX = "GDIndex"
    ONEDRIVE_VERCEL_INDEX = "onedriver-vercel-index"
    GONELIST = "GONEList"
    SHARELIST = "ShareList"
    YUKI_DRIVE = "Yuki Drive"
    ALIST = "AList"
    WATCHLIST_ON_FIRE = "Watchlist on fire 🔥🔥🔥"

File: app/src/main/java/xyz/zimtools/zyod/od/ODType.java
package xyz.zimtools.zyod.od;

public enum ODType {
    GENERIC("Generic"),
    GOINDEX("GoIndex"),
    FODI("FODI"),
    ZFILE("ZFile"),
    GDINDEX("GDIndex"),
    ONEDRIVE_VERCEL_INDEX("onedrive-vercel-index"),
    GONELIST("GONEList"),
    SHARELIST("ShareList"),
    YUKIDRIVE("Yuki Drive"),
    ALIST("AList"),
    WATCHLIST_ON_FIRE("Watchlist on fire 🔥🔥🔥");
    private final String odName;

    ODType(String odName) {
        this.odName = odName;
    }

    @Override
    public String toString() {
        return this.odName;
    }
}